src/main/scala/io/flow/apidoc-json-validation/JsonValidator.scala

```diff
@@ -321,6 +380,81 @@ case class JsonValidator(val service: Service) {
     }
   }
 
+  def validateUuid(prefix: String, js: JsValue): Either[Seq[String], JsString] = {
+    js match {
+      case v: JsArray => {
+        v.value.size match {
+          case 1 => validateUuid(prefix, v.value.head)
+          case _ => Left(Seq(s"$prefix must be a UUID and not an array"))
+        }
+      }
+      case v: JsBoolean => Left(Seq(s"$prefix must be a UUID and not a boolean"))
+      case JsNull => Left(Seq(s"$prefix must be a UUID and not null"))
+      case v: JsNumber => Left(Seq(s"$prefix must be a UUID and not a number"))
+      case v: JsObject => Left(Seq(s"$prefix must be a UUID and not a object"))
+      case v: JsString => {
+        Try {
+          java.util.UUID.fromString(v.value)
+        } match {
+          case Success(v) => Right(JsString(v.toString))
+          case Failure(_) => Left(Seq(s"$prefix must be a valid UUID"))
+        }
+      }
+    }
+  }
+
+  def validateDateIso8601(prefix: String, js: JsValue): Either[Seq[String], JsString] = {
+    js match {
+      case v: JsArray => {
+        v.value.size match {
+          case 1 => validateDateIso8601(prefix, v.value.head)
+          case _ => Left(Seq(s"$prefix must be a valid ISO 8601 date and not an array"))
+        }
+      }
+      case v: JsBoolean => Left(Seq(s"$prefix must be a valid ISO 8601 date and not a boolean"))
+      case JsNull => Left(Seq(s"$prefix must be a valid ISO 8601 date and not null"))
+      case v: JsNumber => Left(Seq(s"$prefix must be a valid ISO 8601 date and not a number"))
+      case v: JsObject => Left(Seq(s"$prefix must be a valid ISO 8601 date and not a object"))
+      case v: JsString => {
+        Try {
+          // Check that the provided value is strictly in yyyy-MM-dd format
+          if(v.value == DateTimeFormat.forPattern("yyyy-MM-dd").print(DateTime.parse(v.value))) {
+            v.value
+          } else {
+            sys.error(s"${v.value} is not in the proper format")
+          }
+        } match {
+          case Success(v) => Right(JsString(v.toString))
+          case Failure(_) => Left(Seq(s"$prefix must be a valid ISO 8601 date"))
+        }
+      }
+    }
+  }
+
+  def validateDateTimeIso8601(prefix: String, js: JsValue): Either[Seq[String], JsString] = {
+    js match {
+      case v: JsArray => {
+        v.value.size match {
+          case 1 => validateDateTimeIso8601(prefix, v.value.head)
+          case _ => Left(Seq(s"$prefix must be a valid ISO 8601 datetime and not an array"))
+        }
+      }
+      case v: JsBoolean => Left(Seq(s"$prefix must be a valid ISO 8601 datetime and not a boolean"))
+      case JsNull => Left(Seq(s"$prefix must be a valid ISO 8601 datetime and not null"))
+      case v: JsNumber => Left(Seq(s"$prefix must be a valid ISO 8601 datetime and not a number"))
+      case v: JsObject => Left(Seq(s"$prefix must be a valid ISO 8601 datetime and not a object"))
+      case v: JsString => {
+        Try {
+          DateTime.parse(v.value)
```

Comment:
let's use:

`ISODateTimeFormat.dateTimeParser.parseDateTime`

to match apidoc clients